### PR TITLE
Scroll Ecosystem Update

### PR DESCRIPTION
**Pull request by dghelm**
Run from internal tracking of hackers and ecosystem projects. This is our first time making a PR from output of this script and datasource, so we'd appreciate any feedback moving forwards.

Subsequent diffs should be easier to read, as the script uses alphabetical ordering.

### CODE REVIEW

**Review by eherrerosj — CHANGES_REQUESTED** [review]
This is off to a good start! There's some work that needs to be done I'm seeing:
1. Some links belong to orgs / users, not repos, make sure you discard user links and move the org links to the `github_organizations` if the entire org builds in Scroll
2. Some repos don't mention Scroll in the code / readme at all, we need a way to make sure the repo uses Scroll

**Comment by dghelm** [comment]
Awesome, we'll get this fixed soon.

As for showing things are built on/deployed to Scroll, should that reasoning be in the PR? And is there a preference on what that reasoning looks like? Deployment config code, documentation, contract address?

**Comment by eherrerosj** [comment]
> Awesome, we'll get this fixed soon.

Thx, I'll keep an eye out.

> As for showing things are built on/deployed to Scroll, should that reasoning be in the PR? And is there a preference on what that reasoning looks like? Deployment config code, documentation, contract address?

There's no need to reason every case. I'd like to understand how you guys are gathering all these repos to validate the approach with you guys and also adjust our repo classifiers / validators for Scroll if needed, otherwise we will see a too high False Positive rate.

**Comment by dghelm** [comment]
>  I'd like to understand how you guys are gathering all these repos to validate the approach with you.

So this list has been curated from 2 internal data sources:

1. Repos of projects that have built on us at hackathons (minimum requirement is a smart contract deployment to Scroll Sepolia testnet or Scroll mainnet, only those confirmed to have deployed included)
2. Orgs and repos of projects that have worked with our ecosystem team and are confirmed as deployed on Scroll or Scroll Sepolia (or in the case of infra, support these networks via providing RPC, indexers, etc.)

I think data source #2 is actually the more confusing one at this point, as we may have included org URLs as repos, and even still, there might not be specific references to Scroll if the project itself is 90% closed source with an open-source website.

My suggestion for moving ahead right now would be to drop source #2 except for projects where it's extremely clear they're a Scroll-native project, and to seek clarify from your team on where we might "draw the line", especially for multi-chain projects.

**Comment by eherrerosj** [comment]
> My suggestion for moving ahead right now would be to drop source B except for projects where it's extremely clear they're a Scroll-native project, and to seek clarify from your team on where we might "draw the line", especially for multi-chain projects.

Agreed with your suggestion. For fairness to the other projects, let's keep only those repos / orgs where there are clear mentions / signals / code referring to Scroll publicly.

Please keep in mind the data freeze happens on December 14th. Do you think you guys will be ready for a new review round by EOW?

**Comment by dghelm** [comment]
Sorry for the delay here @eherrerosj -- I think we need to set up better automation and testing on our end before we pass along data from our Partnerships team.

In an effort to get things resolved more quickly, we've cleaned up our data to only include repos from projects that have explicitly deployed to Scroll or Scroll Sepolia. This data should be exposed either in the linked repos or the projects pages for the hackathons that many projects were built during (not included here, but will be available in another public data source we're looking to release soon).

Thanks for being patient with us while we sort this out!

**Comment by eherrerosj** [comment]
> Sorry for the delay here @eherrerosj -- I think we need to set up better automation and testing on our end before we pass along data from our Partnerships team.
> 
> In an effort to get things resolved more quickly, we've cleaned up our data to only include repos from projects that have explicitly deployed to Scroll or Scroll Sepolia. This data should be exposed either in the linked repos or the projects pages for the hackathons that many projects were built during (not included here, but will be available in another public data source we're looking to release soon).
> 
> Thanks for being patient with us while we sort this out!

sounds good thank you! Could you please solve the issues that the validation step is detecting in the PR?

File: data/ecosystems/s/scroll.toml [hunk 1]
```diff
@@ -6,62 +6,2042 @@ sub_ecosystems = ["Kodadot"]
...
+url = "https://github.com/0xDaenerys/gated-bounty-protocol"
+
+[[repo]]
+url = "https://github.com/0xDononymous/dononymous"
+
+[[repo]]
+url = "https://github.com/0x73696d616f/SwiftGate"
+
+[[repo]]
+url = "https://github.com/0xBeans/Bear-Bonds"
```
[the remark this answers is not in the document]
Comment (dghelm) [reply data/ecosystems/s/scroll.toml:25]: My bad -- messed up my merge conflict resolution, but updated out script to be case-insensitive for sorting, which should make variations of this issue easier to spot in the future.

**Comment by dghelm** [comment]
As far as I can tell, I just messed up the git merge, and the script shouldn't do this again. Modified to be case-insensitive for sorting, which makes catching failed deduping a bit easier as well.

**Review by eherrerosj — CHANGES_REQUESTED** [review]
could you please check the URLs that are added as repos but in reality are org URLs? 

In those cases, please check if the org represents a project building in Scroll:
- If so, then please create a new ecosystem (new toml file) with the org and link the ecosystem to Scroll using the new eco title in the sub_ecosytems field in Scroll
- If not, please only keep the repos that mainly use Scroll

File: data/ecosystems/s/scroll.toml [hunk 2]
```diff
@@ -6,269 +6,2550 @@ sub_ecosystems = ["Kodadot"]
...
+url = "https://github.com/onthisxyz/shortcuts"
+
+[[repo]]
+url = "https://github.com/Open-Sorcerer/scroll-art-"
+
+[[repo]]
+url = "https://github.com/orgs/fight-club-dao/repositories"
+
+[[repo]]
+url = "https://github.com/orgs/hackashell"
```
[the remark this answers is not in the document]
Comment (dghelm) [reply data/ecosystems/s/scroll.toml]: Whoops -- didn't know this was a URL pattern. Fixed.

**Comment by dghelm** [comment]
I know we're getting down to the wire here @eherrerosj -- let me know if there's anything else I can do!

**Comment by eherrerosj** [comment]
> I know we're getting down to the wire here @eherrerosj -- let me know if there's anything else I can do!

I've reviewed the PR. Here are some comments.

### Invalid repo URLs
There are still some invalid URLs:
```
https://github.com/0xLuxora
https://github.com/0xinit/zkMon/tree/frontend-without-history
https://github.com/Ethindia23
https://github.com/clade-club
https://github.com/dharshini1064?tab=repositories
https://github.com/distroinfinity/ethIndia23/tree/main
https://github.com/ethindia-2023
https://github.com/ethindia2023-wagerbox
https://github.com/gaavebase
https://github.com/mannaninamdar
https://github.com/orgs/AlgoCombo/repositories
https://github.com/orgs/DeCarb-ETH/repositories
https://github.com/orgs/Yede-Log/repositories
https://github.com/wayve-ethglobal-ist
```
You can use [my previous suggestion](https://github.com/electric-capital/crypto-ecosystems/pull/658#pullrequestreview-1780498541) to deal with those.

### Repos that might not belong to Scroll
Attached file with a list of 100 repos that might not belong to Scroll. Keep in mind we have a set of heuristics to find Scroll repos and it might not be 100% accurate.

I suggest you take a look at those repos and then remove them from the PR. If you think they belong to Scroll I would appreciate if you can share the places within the code that demonstrate the usage of Scroll. This way I can fix whatever is wrong with out classifiers.


Just so you have an idea of the timeline in our side, we will likely freeze the data at the end of next week so would appreciate if you work on the changes asap, this way we can ensure that your suggested changes are reflected in the 2023 Developer Report.

Thanks!

**Comment by dghelm** [comment]
Latest commits standardize URL structure and fix invalid URLs. 

We've also implemented an exclude list to avoid the "Repos likely not Scroll" as we only found 2 projects that mention Scroll explicitly in their repo, but for your reference:

1. [zka-fi](https://github.com/zka-fi/zkafi/blob/2b36a11f92c4fad5d8831a155164b19cfb4d9143/README.md?plain=1#L26) -- This repo seems to use submodules in an odd way, but following the contracts submodule, you can see the deployment config includes [Scroll testnet](https://github.com/zka-fi/contract/blob/cfef747f110b5438fd8425aa903310f1535326d8/hardhat.config.ts#L21).
2. [Otherswipe](https://github.com/aloksahay/Otherswipe/blob/4edf85fc8e39463a1536a0657a5ff5d1408bf35f/README.md?plain=1#L9) -- They mention their prize which was dependent on a checked deployment, but as it's not implemented in their repo, should be excluded from Dev Report.

**Review by eherrerosj — APPROVED** [review]
awesome work, thank you!

**Pull request merged**